include/cuco/detail/static_map.inl

```diff
@@ -123,18 +123,71 @@ void static_map<Key, Value, Scope, Allocator>::contains(
 }
 
 template <typename Key, typename Value, cuda::thread_scope Scope, typename Allocator>
-template <typename Hash, typename KeyEqual>
-__device__ bool static_map<Key, Value, Scope, Allocator>::device_mutable_view::insert(
-  value_type const& insert_pair, Hash hash, KeyEqual key_equal) noexcept
+template <typename pair_type, typename Hash, typename KeyEqual>
+__device__ std::enable_if_t<cuco::detail::is_packable<pair_type>(), bool>
+static_map<Key, Value, Scope, Allocator>::device_mutable_view::insert(value_type const& insert_pair,
+                                                                      Hash hash,
+                                                                      KeyEqual key_equal) noexcept
 {
   auto current_slot{initial_slot(insert_pair.first, hash)};
 
   while (true) {
     using cuda::std::memory_order_relaxed;
     auto expected_key   = this->get_empty_key_sentinel();
     auto expected_value = this->get_empty_value_sentinel();
-    auto& slot_key      = current_slot->first;
-    auto& slot_value    = current_slot->second;
+
+    key_type const existing_key = current_slot->first.load(memory_order_relaxed);
+    // the key we are trying to insert is already in the map, so we return with failure to insert
+    if (key_equal(existing_key, insert_pair.first)) { return false; }
+
+    cuco::detail::pair_converter<pair_type> converter;
+
+    converter.pair =
+      cuco::make_pair<Key, Value>(std::move(expected_key), std::move(expected_value));
+    auto empty_sentinel = converter.packed;
+    converter.pair      = insert_pair;
+    auto tmp_pair       = converter.packed;
+
+    auto slot = reinterpret_cast<
+      cuda::atomic<typename cuco::detail::pair_converter<pair_type>::packed_type>*>(current_slot);
+
+    bool success = slot->compare_exchange_strong(empty_sentinel, tmp_pair, memory_order_relaxed);
+    converter.packed   = empty_sentinel;
+    auto expected_pair = converter.pair;
```

Comment:
Reusing the same `pair_converter` object is harder to follow. Just use separate objects.